File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.java

import java.util.HashSet;

class Solution {
    public int lengthOfLongestSubstring(String s) {
        int n = s.length();

        HashSet<Character> set = new HashSet<>();
        int left = 0, right = 0, max = 0;

        while (right < n) {
            if (!set.contains(s.charAt(right))) {
                set.add(s.charAt(right));
                right++;
                max = Math.max(max, right - left);
            } else {
                set.remove(s.charAt(left));
                left++;
            }
        }

        return max;
    }
}

File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        unordered_map<char,int> mp;
        int mx=0;
        int i=0;
        for(int j=0;j<s.size();j++)
        {
            mp[s[j]]++;
            while(mp[s[j]]>=2){mp[s[i]]--;i++;}
            mx=max(mx,j-i+1);
        }
        return mx;
    }
};